lambda_code/src/pds_service.py

```diff
@@ -0,0 +1,71 @@
+import base64
+import json
+import time
+import uuid
+
+import jwt
+import requests
+
+
+class Authenticator:
+    def __init__(self, secret_manager_client, environment):
+        self.secret_manager_client = secret_manager_client
+        # FIXME: the secret name's env is hardcoded. AMB-1838 should address this.
+        self.secret_name = 'imms/pds/int/jwt'
+        self.token_url = f"https://{environment}.api.service.nhs.uk/oauth2/token"
+
+    def get_pds_secrets(self):
+        kwargs = {"SecretId": self.secret_name}
+        response = self.secret_manager_client.get_secret_value(**kwargs)
+        secret_object = json.loads(response['SecretString'])
+        secret_object['private_key'] = base64.b64decode(secret_object['private_key']).decode()
+
+        return secret_object
+
+    def create_jwt(self):
+        secret_object = self.get_pds_secrets()
+        current_timestamp = int(time.time())
+        claims = {
+            "iss": secret_object['api_key'],
+            "sub": secret_object['api_key'],
+            "aud": self.token_url,
+            "iat": current_timestamp,
+            "exp": current_timestamp + 30,
```

Comment:
Is a 30 second expiry intended?

Reply:
I will address this separately in AMB-1839 because that cache logic will deal with expiry time.